src/teleop_twist_joy.cpp

```diff
@@ -153,6 +153,114 @@ TeleopTwistJoy::TeleopTwistJoy(const rclcpp::NodeOptions& options) : Node("teleo
   }
 
   pimpl_->sent_disable_msg = false;
+
+  auto param_callback = 
+  [this](std::vector<rclcpp::Parameter> parameters)
+  {
+    auto result = rcl_interfaces::msg::SetParametersResult();
+    result.successful = true;
+
+    // Loop to check if changed parameters are of expected data type
+    for(const auto & parameter : parameters)
```

Comment:
The reason I have two different loops is that if we do type checking and assigning parameter value to the member variable simultaneously, there might come an instance where a particular parameter does not match the type specifications, and the whole parameter-change "result.successful" will be equal to "false". This means that the parameter values will not change, but we would have modified the values of the member functions already to reflect the change. 
This is why I kept it in two different loops, so that assigning values to the member variables happen only if ALL parameter checks are satisfied.
Let me know what you think

Reply:
That makes sense, thanks for explaining.